src/codegen/codegen_llvm.cpp

```diff
@@ -390,12 +601,49 @@ void CodeGen_LLVM::visit(const Call* op) {
 
 void CodeGen_LLVM::visit(const IfThenElse* op) {
   auto _ = CodeGen_LLVM::IndentHelper(this, "IfThenElse");
-  throw logic_error("Not Implemented for IfThenElse.");
+
+  // Create the BasicBlocks
+  auto* true_bb = llvm::BasicBlock::Create(*this->Context, "true_bb", this->Func);
+  auto* false_bb = llvm::BasicBlock::Create(*this->Context, "false_bb", this->Func);
+  auto* after_bb = llvm::BasicBlock::Create(*this->Context, "after_bb", this->Func);
+
+  // Create condition
+  Builder->CreateCondBr(codegen(op->cond), true_bb, false_bb);
+
+  // True case
+  Builder->SetInsertPoint(true_bb);
+  codegen(op->then);
+  Builder->CreateBr(after_bb);
+
+  // False Case
+  Builder->SetInsertPoint(false_bb);
+  if (op->otherwise != nullptr) {
+    codegen(op->otherwise);
+  }
+  Builder->CreateBr(after_bb);
+
+  // Set the Insertion point to the next BasicBlock
+  Builder->SetInsertPoint(after_bb);
 }
 
-void CodeGen_LLVM::visit(const Case* op) {
+namespace {
+Stmt caseToIfThenElse(std::vector<std::pair<Expr, Stmt>> clauses, bool alwaysMatch) {
+  std::vector<std::pair<Expr, Stmt>> rest(clauses.begin() + 1, clauses.end());
+  if (rest.size() == 0) {
+    // if alwaysMatch is true, then this one goes into the else clause,
+    // otherwise, we generate an empty else clause
+    return !alwaysMatch ? clauses[0].second
+                        : IfThenElse::make(clauses[0].first, clauses[0].second, Comment::make(""));
+  } else {
+    return IfThenElse::make(
+        clauses[0].first, clauses[0].second, caseToIfThenElse(rest, alwaysMatch));
+  }
+}
+}  // anonymous namespace
+
+void CodeGen_LLVM::visit(const Case* op) {  // Not tested
```

Comment:
Given we don't have any tests for this opcode, let's postpone it to another PR. Just comment the call to `codegen(caseToIfThenElse(...))` and throw an error.

Reply:
Actually, this call is essential to series of IfThenElse!
This is not the `case` used for Switch, this is used to generate a series of conditionals like [here](https://github.com/Robertorosmaninho/taco/blob/4afa91a6667be480ee05de2f8ee1bb746d909b36/include/taco/ir/ir.h#L574)!

Reply:
Ok! But do we have any test the exercise the code we generate?